### ISSUE TITLE
[Feature] Time axis interval

### ISSUE THREAD

**Issue by Ovilia** [apache/echarts#17576]
### What problem does this feature solve?

Since [v5](https://github.com/apache/echarts/pull/12859), the `interval` option of time axes is no long valid because the span of each two ticks are not always the same (e.g., Feburary has 28/29 days while March has 31 days). But it's a common requirement to set the interval level (or the minimium interval level) to be yearly/monthly/dayly/... and it can hardly be done with the current API.

### What does the proposed API look like?

I'd suggest we support `interval`, `minInterval`, and `maxInterval` in the type of [`TimeUnit`](https://github.com/apache/echarts/blob/6ee5f7076d84b72dc65b4c53e92e5003b8624b2f/src/util/time.ts#L58).

> The reason why `number` type should not be used instead is that each month has different days so using the interval of `1000 * 3600 * 24 * 30` is not accurate and unreasonable.

For example, to display use ticks by month:

```ts
xAxis: {
  type: 'time',
  interval: 'month',
  axisLabel: { // optional
    showMinLabel: true,
    formatter: {
      year: '{yyyy}',
      month: '{MMM}',
      day: '{yyyy} {MMM}'' // hack for the first day
    }
  }
}
```

The ticks will be at the first date of each month (excluding the first one because it's displaying only because `showMinLabel: true`):

```
    |_____________|__________________________|__________________________|__ ...
    2019 Dec      2020                       Feb                        March
```

`minInterval` and `maxInterval` works similarly.

**Comment by pissang** [comment]
Sounds like a intuitionistic design. But there are two things we should always consider when designing this kind of domain-specific language:

1. We should be careful about the code complexity parsing the language. Including handling all kind of edge cases.
2. It can't benifit from TypeScript. Developers may be easy to have typo in it and spending hours to figure out what's wrong.

**Comment by Ovilia** [comment]
@pissang Yes, I've considered about the second problem. Another option would be to use `intervalUnit: 'day', interval: 7`, which may not be as intuitive as `'7 days'` but more rebust. Which one do you think is better?

**Comment by StimioAnthonyLeRoy** [comment]
@Ovilia You think this feature will be available for the next version ? 
Because I have to stay with the V4 version because of this problem.

**Comment by Ovilia** [comment]
@StimioAnthonyLeRoy This feature may probably go with 5.5.0 because it's a major feature. And our next version 5.4.1 is scheduled at Novermber. So You may proabably wait until early 2023 for this feature. However, if it's OK to use nightly build, hopefully it can be ready by the end of next month.

**Comment by pAfW001** [comment]
@Ovilia 
Hi, I think my problem might be fixed with the next release, but I'm not really sure, so pls let me explain here.
I'm using version 5.3.2
Sample code: https://jsfiddle.net/rpgeyw7b/

Data is in ISO format but depending on certain conditions, I need to display the axis labels in either YYYY-MM or YYY-MM-DD.
In the sample code, the xAxis ticks and labels are not correctly converted (Feb is missing).

Please confirm if this issue would be fixed in the next release 5.5.0
Thank you!

**Comment by Ovilia** [comment]
@pAfW001 Feb is not displayed because it overlapps with the previous labels. This is the expected behavior.

**Comment by pAfW001** [comment]
@Ovilia Thank you for your reply!
If that is the case, then axisLabel should be empty there.. Could you please have a look at the **xAxis** of the chart again ?
The problem is that March is displayed in the position of Feb, causing all other xAxis labels to be shifted to the left (1-month gap). ALso, there are only 11 xAxis ticks even though there are 12 data points

**Comment by Ovilia** [comment]
@pAfW001 Yes, this seems to be a bug. I will check this later when this PR is about to be finished.

**Comment by cllee1214** [comment]
it's looks like do not work in **wechat minprogram**.my version is  5.4.2.
the code as follows:
![1687945093588](https://github.com/apache/echarts/assets/8358471/0d24e5f9-624b-483d-bf77-390e134fe42d)

or

![image](https://github.com/apache/echarts/assets/8358471/223230a1-0af1-4d14-86b2-c586023e9f32)

**Comment by fasergeyte** [comment]
Hi guys!, Will it be available soon? And another question, is there any workaround how I could achieve same functionality right now?

**Comment by kupchic** [comment]
> Sounds like a intuitionistic design. But there are two things we should always consider when designing this kind of domain-specific language:
> 
> 1. We should be careful about the code complexity parsing the language. Especially when handling all kind of edge cases.
> 2. It can't benifit from TypeScript. Developers may be easy to have typo in it and spending hours to figure out what's wrong.

developers spending hours to set correct interval rules for time axes with current API

**Comment by shadow-identity** [comment]
@Ovilia 
Hi, could you please merge it? So other developers does not waste hours of work, like me, trying to make work things that are not implemented.
